contracts/base/sushi-base/MasterChefStrategyUL.sol

```diff
@@ -0,0 +1,391 @@
+pragma solidity 0.5.16;
+pragma experimental ABIEncoderV2;
+
+import "@openzeppelin/contracts/math/Math.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
+import "@openzeppelin/contracts/token/ERC20/SafeERC20.sol";
+
+import "./interfaces/IMasterChef.sol";
+import "../interface/IStrategy.sol";
+import "../interface/IVault.sol";
+import "../interface/uniswap/IUniswapV2Pair.sol";
+import "../interface/uniswap/IUniswapV2Router02.sol";
+
+import "../upgradability/BaseUpgradeableStrategyUL.sol";
+
+contract MasterChefStrategyUL is IStrategy, BaseUpgradeableStrategyUL {
+    using SafeMath for uint256;
+    using SafeERC20 for IERC20;
+
+    // config parameters for BaseUpgradebaleStrategyUL
+    // the constant values will be replaced during compile time, so no GAS issues during deploy/runtime
+    // but makes is easier to read/change parameters
+    address private constant UL_REGISTRY =
```

Comment:
I think putting these parameters here could lead to problems when upgrading the contract when these parameters are not put in exactly the same order in the new upgraded implementation. This might not be a huge problem, but by inputting them directly in the initialize function this risk is taken away.

Reply:
The occurrences of the constant variables are replaced by the assigned expression during compile time. So no storage slots are used and therefore would have no impact on the storage layout (no impact on upgradeability). 
But I changed it nevertheless and used named arguments to make things more readable.